syntax = "proto3";

service SubmissionManager {
    rpc Submit(SubmitRequest) returns (SubmitResponse) {}
    rpc GetStorageManagerConfig(GetStorageManagerConfigRequest) returns (GetStorageManagerConfigResponse) {}
    rpc SimulationSubmit(SimulationSubmitRequest) returns (SimulationSubmitResponse) {}
}

message SubmitRequest {
    string working_dir = 1;
}

message SubmitResponse {
    bool success = 1;
    int32 job_id = 2;
}

message GetStorageManagerConfigRequest {

}

message GetStorageManagerConfigResponse {
    bool success = 1;
    string class_name = 2;
    map<string, string> args = 3;
}

message Description {
    map<string, string> description = 1;
}

message SimulationSubmitRequest {
    Description job_description = 1;
    repeated Description task_descriptions = 2;
}

message SimulationSubmitResponse {
    bool success = 1;
}